### ISSUE TITLE
False-positive Style/Documentation offence on trailing comments

### ISSUE THREAD

**Issue by RGBD** [bbatsov/rubocop#1905]
```shell
% rubocop -V
0.31.0 (using Parser 2.2.2.3, running on ruby 2.2.1 x86_64-linux)
```

### tl;dr

[This](https://gist.github.com/RGBD/0fc6e0377fdefbbf721b#file-minimal-rb) file:

```shell
% cat minimal.rb
module M
  # top level class documentation
  class C
	def foo; end
  end # class ending comment, the troublemaker
end
```

Produces this:

```shell
% rubocop minimal.rb
Inspecting 1 file
C

Offenses:

minimal.rb:3:3: C: Missing top-level class documentation comment.
  class C
  ^^^^^

1 file inspected, 1 offense detected
```

### Problem domain

I've seen people use end block comment do denote which class it ends. The style
guide indirectly discourages need of such comments [here](https://github.com/bbatsov/ruby-style-guide#file-classes), and maybe it
even deserves a separate cop. But because of this issue the user is provided
with irrelevant problem description and location.

### Why this happens?

There is a little ambiguity on how trailing comments should be treated by [parser](https://github.com/whitequark/parser/). In
particular, Parser::Source::Comment associates comments after the last child of
a node with this last child. So, when Rubocop::Cop::Style::Documentation parses
last class enclosed in module, it gives unexpected results, see [gist](https://gist.github.com/RGBD/0fc6e0377fdefbbf721b#file-default-log).

### Possible fix

Cop could check for associated comments only before class/module definition.
See my PR (#1906).

### The reason lies deeper

I think, the behaviour of parser comment association should be changed to be
more intuitive in this case. I've found example for false-positive match.
See this [gist](https://gist.github.com/RGBD/0fc6e0377fdefbbf721b#file-default-log).

### Summary

If my approach seems good to you, I'll add rspec examples so we can merge. I
hope @whitequark will join this conversation, because there are even more edge
cases, that should be fixed in parser itself, not in rubocop.

**Comment by bbatsov** [comment]
Guess we need a new parser release from @whitequark, so you can be able to update this PR according to the changes you made there.

**Comment by RGBD** [comment]
@bbatsov, my PR doesn't depend on parser changes, the fix will work with the old parser version too.
The only difference is that I add test, that will start failing without m
